src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLGuid.cs

```diff
@@ -125,16 +121,22 @@ public static SqlGuid Parse(string s)
         // Comparison operators
         private static EComparison Compare(SqlGuid x, SqlGuid y)
         {
+            // Comparison orders.
+            ReadOnlySpan<int> rgiGuidOrder = new int[16] { 10, 11, 12, 13, 14, 15, 8, 9, 6, 7, 4, 5, 0, 1, 2, 3 };
```

Comment:
Nit: We could make this a `ROS<byte>` and check it in now, and we wouldn't have to worry about downlevel weirdness or compiler support.